### DISCUSSION TITLE
Problem with supersampling_factor = 3

### DISCUSSION THREAD

**Discussion by ZareefJafar** [AstroBridge/BDLensing#235] [Q&A]
As discussed in last meeting with @rafee-adnan and @Rajit13  I changed supersampling_factor to **3** from **1** in Numerical settings for PSO of **lens_model_DESIJ1554+0443**

`kwargs_numerics = {"supersampling_factor": 3, "supersampling_convolution": False}`

I kept everything else unchanged. But there was a significant change in reconstructed images.

Best fit values and reconstructed images with "supersampling_factor": 1,

```
{'kwargs_lens': [{'theta_E': 1.5049071454942418, 'gamma': 2, 'e1': -0.03142944020111962, 'e2': -0.09799250158910201, 'center_x': -0.30819684986594775, 'center_y': -0.13848398895797967}, {'gamma1': 0.11783022807885163, 'gamma2': 0.04206611826391211, 'ra_0': 0, 'dec_0': 0}], 'kwargs_source': [{'amp': 1, 'R_sersic': 0.0688443992940853, 'n_sersic': 1.0, 'e1': -0.4501990603846126, 'e2': -0.07459561772663237, 'center_x': -0.5482959649040664, 'center_y': -0.15937736381538095}, {'amp': 1, 'n_max': 5, 'beta': 0.1499784335589324, 'center_x': -0.5482959649040664, 'center_y': -0.15937736381538095}], 'kwargs_lens_light': [{'amp': 1, 'R_sersic': 0.8717125354390587, 'n_sersic': 4.0, 'e1': -0.023419404461170634, 'e2': -0.055394182720308896, 'center_x': -0.329654972248968, 'center_y': -0.09663526349711914}, {'amp': 1, 'R_sersic': 0.8155529922878545, 'n_sersic': 4.0, 'e1': 0.46227750962172076, 'e2': -0.12111881054760087, 'center_x': -0.329654972248968, 'center_y': -0.09663526349711914}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}\n-1.0020587359779218 reduced X^2 of all evaluated imaging data combined (without degrees of freedom subtracted).\nreduced chi^2 of data  0 =  1.0020587359779216\n
```
![image](https://github.com/ajshajib/BDLensing/assets/49613484/d50b695e-95b2-4d5b-96da-c907d371b5fa)
![image](https://github.com/ajshajib/BDLensing/assets/49613484/e3d5a28a-7b2e-4d3e-80b3-8904c5028047)

Best fit values and reconstructed images with "supersampling_factor": 3,
```
{'kwargs_lens': [{'theta_E': 1.3, 'gamma': 2, 'e1': -0.041, 'e2': -0.095, 'center_x': -0.2939, 'center_y': -0.114}, {'gamma1': 0.1035, 'gamma2': 0.0391, 'ra_0': 0, 'dec_0': 0}], 'kwargs_source': [{'amp': 1, 'R_sersic': 0.1, 'n_sersic': 1.0, 'e1': -0.4998, 'e2': -0.22137, 'center_x': -0.27066, 'center_y': -0.164}, {'amp': 1, 'n_max': 5, 'beta': 0.145, 'center_x': -0.27066, 'center_y': -0.164}], 'kwargs_lens_light': [{'amp': 1, 'R_sersic': 0.4761, 'n_sersic': 4.0, 'e1': 0.2178, 'e2': -0.0474, 'center_x': 0.0295, 'center_y': -0.2483}, {'amp': 1, 'R_sersic': 0.4761, 'n_sersic': 4.0, 'e1': 0.2178, 'e2': -0.0474, 'center_x': 0.0295, 'center_y': -0.2483}], 'kwargs_ps': [], 'kwargs_special': {}, 'kwargs_extinction': []}
-980.7412336204648 reduced X^2 of all evaluated imaging data combined (without degrees of freedom subtracted).
reduced chi^2 of data  0 =  980.7412336204648
```

![image](https://github.com/ajshajib/BDLensing/assets/49613484/1a88f98a-afe4-413e-9e73-bd0902dbacb4)

![image](https://github.com/ajshajib/BDLensing/assets/49613484/687c027c-32e8-4205-84c4-c81a9da720f9)

**Comment by ZareefJafar** [comment]
Found the solution.

In the kwargs_likelihood of Numerical settings the` "check_positive_flux"` is True. Making it `False` solves the problem. Here is my finale Numerical settings:

```
kwargs_likelihood = {"check_bounds": True, 
                     "image_likelihood_mask_list": [mask_img],
                     "check_positive_flux": False}

kwargs_numerics = {"supersampling_factor": 3, "supersampling_convolution": False}
```

**Comment by ajshajib** [comment]
It should be fine to run PSO only with `supersampling_factor = 1`. However, it should be set to 3 for the last MCMC run, which would provide numbers to go into the paper, as this provides better accuracy. Setting `check_positive_flux = False` can be ok, unless it has other undesirable consequences, for example, making one of the Sersic profile amplitudes negative, so I would check the `amp` values in `kwargs_result` before accepting `check_positive_flux = False`.
